modules/media_riddle_marketplace/src/Plugin/MediaEntity/Type/Riddle.php

```diff
@@ -163,27 +177,27 @@ public function getField(MediaInterface $media, $name) {
           return FALSE;
 
         case 'thumbnail_local':
-          $local_uri = $this->getField($media, 'thumbnail_local_uri');
-          if ($local_uri) {
-            if (file_exists($local_uri)) {
-              return $local_uri;
-            }
-            else {
-
-              $directory = dirname($local_uri);
-              file_prepare_directory($directory, FILE_CREATE_DIRECTORY | FILE_MODIFY_PERMISSIONS);
-
-              $image_url = $this->getField($media, 'thumbnail');
-
-              return file_unmanaged_save_data(file_get_contents($image_url), $local_uri, FILE_EXISTS_REPLACE);
-            }
-          }
-          return FALSE;
-
         case 'thumbnail_local_uri':
           if (isset($riddle['image'])) {
-            return $this->configFactory->get('media_riddle_marketplace.settings')
-              ->get('local_images') . '/' . $code . '.' . pathinfo(parse_url($riddle['image'], PHP_URL_PATH), PATHINFO_EXTENSION);
+            $directory = $this->configFactory->get('media_riddle_marketplace.settings')->get('local_images');
+            // Try to load from local filesystem.
+            $absolute_uri = current(glob($this->fileSystem->realpath($directory . '/' . $code . ".*{jpg,jpeg,png,gif}"), GLOB_BRACE));
+            if ($absolute_uri) {
+              return $directory . '/' . $this->fileSystem->basename($absolute_uri);
+            }
+            file_prepare_directory($directory, FILE_CREATE_DIRECTORY | FILE_MODIFY_PERMISSIONS);
+            // Get image from remote and save locally.
+            try {
+              $response = \Drupal::httpClient()->head($riddle['image']);
+              $format = ExtensionGuesser::getInstance()
```

Comment:
You can use a drupal core service for that, either file.mime_type.guesser.extension or file.mime_type.guesser